modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -597,6 +643,33 @@ void ONNXCompiled::setInput(int i, const cv::Mat &m) {
     preprocess(m, in_tensor_info[ort_idx], in_data[in_idx]);
 }
 
+void ONNXCompiled::setInput(ONNXCallContext &ctx,
+                            const int in_idx,
+                            const int name_idx,
+                            Views& views,
+                            const cv::Rect& roi = cv::Rect()) {
+    const auto in_name = params.input_names[name_idx];
+    const auto ort_idx = getIdxByName(in_tensor_info, in_name);
+
+    switch (ctx.in_shapes[in_idx]) {
+        case cv::GShape::GFRAME: {
+            const cv::MediaFrame& frame = ctx.inFrame(in_idx);
+            views.emplace_back(new cv::MediaFrame::View(frame.access(cv::MediaFrame::Access::R)));
```

Comment:
Make sure your `views` is preallocated to the right size.

Reply:
Added assert.

Reply:
it should be a `reserve()` somewhere earlier in the stack.

the idea is to avoid vector reallocations if possible, but since elements will be moved instead of copied, it should still work for unique pointers over views.